### PR TITLE
DCKUBE-390: Examples of production values.yaml files

### PR DESCRIPTION
**Pull request by errcode1202**
Example "production" values populated with details relating to prerequisites

### CODE REVIEW

File: docs/examples/values/prod/bitbucket.yaml [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+    # -- Specifies the name of the ClusterRoleBinding that will be created if the "serviceAccount.clusterRoleBinding.create" flag is set
+    # If not specified, a name will be auto-generated.
+    name:
+    # -- true if a ClusterRoleBinding should be created, or false if it already exists
+    create: true
+
+database:
+  # -- The JDBC URL of the database to be used by Bitbucket, e.g. jdbc:postgresql://host:port/database
+  # If not specified, then it will need to be provided via the browser during initial startup.
+  url: jdbc:postgresql://bitbucket-postgres-2.c8hzuxrqaglj.eu-west-1.rds.amazonaws.com/bitbucket
```
Comment (nanux) [inline docs/examples/values/prod/bitbucket.yaml:35]: I think we should make this more generic:
```
url: jdbc:postgresql://postgres.k8s.com:5432/bitbucket
```
We could potentially add also examples for other DBs so people don't need to go to documentation. The other examples can be just commented out.

File: docs/examples/values/prod/bitbucket.yaml [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+    # -- The port on which the Bitbucket Kubernetes service will listen
+    port: 80
+    # -- The type of Kubernetes service to use for Bitbucket
+    type: ClusterIP
+    annotations: {}
+  # -- Enable or disable an additional service for exposing SSH for external access.
+  # Disable when the SSH service is exposed through the ingress controller, or enable if the ingress controller does
+  # not support TCP.
+  sshService:
+    enabled: false
```
Comment (nanux) [inline docs/examples/values/prod/bitbucket.yaml:60]: Should this be enabled?

File: docs/examples/values/prod/bitbucket.yaml [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+
+  clustering:
+    # -- Set to true if Data Center clustering should be enabled
+    # This will automatically configure cluster peer discovery between cluster nodes.
+    enabled: true
+
+  elasticSearch:
+    # -- The base URL of the external ElasticSearch instance to be used.
+    # If this is defined, then Bitbucket will disable its internal ElasticSearch instance.
+    baseUrl:
```
Comment (nanux) [inline docs/examples/values/prod/bitbucket.yaml:107]: Elasticsearch is quite important part, we should provide example for it.

File: docs/examples/values/prod/bitbucket.yaml [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+podAnnotations: {}
+#  "name": "value"
+
+volumes:
+  localHome:
+    persistentVolumeClaim:
+      # -- If true, then a PersistentVolumeClaim will be created for each local-home volume.
+      create: true
+      # -- Specifies the name of the storage class that should be used for the local-home volume claim.
+      storageClassName: "ebs-sc"
```
Comment (nanux) [inline docs/examples/values/prod/bitbucket.yaml:240]: Is this a standard name?

File: docs/examples/values/prod/bitbucket.yaml [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+    mountPath: "/var/atlassian/application-data/bitbucket"
+  sharedHome:
+    persistentVolume:
+      # -- If true, then a PersistentVolume will be created for the shared-home volume.
+      create: false
+      # -- Addtional options used when mounting the volume
+      mountOptions: []
+      nfs:
+        # -- The address of the NFS server. It needs to be resolveable by the kubelet, so consider using an IP address.
+        server: ""
```
Comment (nanux) [inline docs/examples/values/prod/bitbucket.yaml:257]: We need to provide example for this as EFS is not supported.

File: docs/examples/values/prod/bitbucket.yaml [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+      persistentVolumeClaim:
+        claimName: "efs-pvc" 
+    # -- Specifies the path in the Bitbucket container to which the shared-home volume will be mounted.
+    mountPath: "/var/atlassian/application-data/shared-home"
+    # -- Specifies the sub-directory of the shared-home volume that will be mounted in to the Bitbucket container.
+    subPath:
+    nfsPermissionFixer:
+      # -- If enabled, this will alter the shared-home volume's root directory so that Bitbucket can write to it.
+      # This is a workaround for a Kubernetes bug affecting NFS volumes: https://github.com/kubernetes/examples/issues/260
+      enabled: false
```
Comment (nanux) [inline docs/examples/values/prod/bitbucket.yaml:283]: Is this required most of the time?

File: docs/examples/values/prod/bitbucket.yaml [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+# Due to the performance requirements it is highly recommended to run all Bitbucket pods
+# in the same availability zone as your dedicated NFS server. To achieve this, you
+# can define `affinity` and `podAffinity` rules that will place all pods into the same zone,
+# and therefore minimise the real distance between the application pods and the shared storage.
+# More specific documentation can be found in the official Affinity and Anti-affinity documentation:
+#  https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity
+#
+# This is an example on how to ensure the pods are in the same zone as NFS server that is labeled with `role=nfs-server`:
+#
+#   podAffinity:
```
Comment (nanux) [inline docs/examples/values/prod/bitbucket.yaml:311]: PodAffinity is required for NFS and Bitbucket app for good performance.

File: docs/examples/values/prod/confluence.yaml [hunk 8]
```diff
@@ -0,0 +1,375 @@
...
+      stackSize: "2048k"
+    container: 
+      requests:
+        cpu: "2"
+        memory: "2.5G"
+  # -- The base URL of the Synchrony service.
+  # This will be the URL that users' browsers will be given to communicate with Synchrony, as well as the URL that the
+  # Confluence service will use to communicate directly with Synchrony, so the URL must be resovable both from inside and
+  # outside the Kubernetes cluster.
+  ingressUrl: confluence-synchrony.k8s.com
```
Comment (nanux) [inline docs/examples/values/prod/confluence.yaml:202]: Does this work without an additional DNS setup? I thought it is usually `confluence-ingress/synchrony` e.g.: https://confluence.k8s.com/synchrony

**Comment by nanux** [comment]
What are our options to keep this in sync with any changes to `values.yaml` default files?

**Review by errcode1202 — COMMENTED** [review]
> What are our options to keep this in sync with any changes to `values.yaml` default files?

I'm not too sure how this can be done effectively without being scripted. This is something to keep in mind, any changes made in the main values.yaml will need to be reflected in these examples which aint great. Im now having reservations about whether this work is actually needed without first getting some customer feedback.

File: docs/examples/values/prod/bitbucket.yaml (hunk 3 above)
Comment (nanux) [reply docs/examples/values/prod/bitbucket.yaml:107]: Yes, I think we have just forgotten to add it to prerequisites. I think it is quite an important part of the BB experience.

**Comment by errcode1202** [comment]
Closing in favour of refactoring product values.yaml's so that they are easier to reason with.

**Pull request closed without merging**